crates/astria-sequencer/src/state_ext.rs

```diff
@@ -167,11 +179,31 @@ pub(crate) trait StateReadExt: StateRead {
 
 impl<T: StateRead> StateReadExt for T {}
 
+fn revision_number_from_chain_id(chain_id: &str) -> u64 {
+    let parts = chain_id.split("-").collect::<Vec<_>>();
```

Comment:
You don't need this allocation, you can use this iterator adaptor: https://doc.rust-lang.org/std/iter/trait.Iterator.html#method.last

Your line below then becomes:

```rust
if let Some(rev_number) = chain_id.split("-").last() {
```

Reply:
removed in favour of regex